pkg/defaults/defaults_test.go

```diff
@@ -418,74 +418,27 @@ func TestStepConfigsForBuild(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			if configs := stepConfigsForBuild(testCase.input, testCase.jobSpec); !stepListsEqual(configs, testCase.output) {
-				t.Logf("%s", diff.ObjectReflectDiff(testCase.output, configs))
-				t.Errorf("incorrect defaulted step configurations,\n\tgot:\n%s\n\texpected:\n%s", formatSteps(configs), formatSteps(testCase.output))
```

Comment:
The output of this was pretty much useless and more confusing than anything else. I've built a pretty dumb sorter for this which allows us to generate a proper diff (and saves a lot of code)